Chess/src/chess/BoardState.java

```diff
@@ -0,0 +1,44 @@
+package chess;
+
+import java.util.Arrays;
+
+public class BoardState {
+
+	private Cell[][] state;
+	
+	public BoardState(Cell[][] state) {
+		this.state = state;
+	}
+	
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof BoardState)) return false;
+        BoardState that = (BoardState) o;
+        return Arrays.deepEquals(this.state, that.state);
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.deepHashCode(this.state);
+    }
+	
+    public void printState() {
+    	for (int i = 0; i < 8; i++) {
+    		for (int j = 0; j < 8; j++) {
+    			System.out.print(state[i][j].getpiece() != null ? state[i][j].getpiece().getPieceType() : "X" );
+    		}
+    		System.out.print("\n");
+    	}
+    }
+    
+    public String buildString() {
+        StringBuilder stateString = new StringBuilder();
+        for (int i = 0; i < 8; i++) {
+            for (int j = 0; j < 8; j++) {
+                stateString.append(state[i][j].getpiece() != null ? state[i][j].getpiece().getPieceType() : "X");
```

Comment:
A bit of a nitpick but I'm not a huge fan of having two characters for each piece present and only one character for every missing piece - could you instead append "XX" so that all state strings are 128 characters?